File: 111術科模擬試題/B1.py
#111商業技藝競賽模擬試題
#Problem #B1 好友滿天下
#Author: Yotrew Wing
#2022/10/17
#https://github.com/yotrew/commercial_skill_competition

n=int(input())
#stat=[{"",set()} for i in range(n)]
stat=dict()
for i in range(n):
    
    ins=input().split(" ",1)
    if not ins[0] in stat:
        stat[ins[0]]=[ins[1]]
    else:
        if ins[1] not in stat[ins[0]]:
            stat[ins[0]].append(ins[1])
stat_key=sorted(stat)
for i in stat_key:
    print(i,len(stat[i]))

File: 111術科模擬試題/B1.cpp
/*
#111商業技藝競賽模擬試題
#Problem #B1 好友滿天下
#Author: Yotrew Wing
#2022/10/17
#https://github.com/yotrew/commercial_skill_competition
*/
#include <iostream>
#include <map>
#include <set>
using namespace std;

int main()
{
    int n;
    cin>>n;
    string ins;
    string delimiter = " ";
    string token="";
    size_t end;
    size_t start=0;
    string country="";
    map<string, set<string>> dic;
    cin.get();
    for(int i=0;i<n;i++){
        getline(cin,ins);
        //cout<<"test"<<ins<<endl;
        start=0;
        end=ins.find(delimiter);
        country=ins.substr(start, end-start);
        token=ins.substr(end+delimiter.size());
        dic[country].insert(token);
    }
    for (auto &item :dic) {
        cout << item.first << " " << item.second.size() << endl;
    }
    return 0;
}
